File: backend/team6/src/test/java/programmers/team6/mock/VacationGrantRuleFinderFake.java
package programmers.team6.mock;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import programmers.team6.domain.vacation.rule.VacationGrantRule;
import programmers.team6.domain.vacation.rule.VacationGrantRuleFinder;
import programmers.team6.domain.vacation.rule.VacationGrantRules;

public class VacationGrantRuleFinderFake extends VacationGrantRuleFinder {

	private List<VacationGrantRule> rules;

	public VacationGrantRuleFinderFake(VacationGrantRule... rules) {
		this.rules = new ArrayList<>(Arrays.asList(rules));
	}

	@Override
	public VacationGrantRules findAll() {
		return new VacationGrantRules(rules);
	}
}

File: backend/team6/src/test/java/programmers/team6/mock/VacationGrantRuleFinderFake.kt
package programmers.team6.mock

import programmers.team6.domain.vacation.rule.VacationGrantRule
import programmers.team6.domain.vacation.rule.VacationGrantRuleFinder
import programmers.team6.domain.vacation.rule.VacationGrantRules
import java.util.*

class VacationGrantRuleFinderFake(vararg rules: VacationGrantRule) : VacationGrantRuleFinder() {
    private val rules: List<VacationGrantRule> =
        ArrayList(Arrays.asList(*rules))

    override fun findAll(): VacationGrantRules {
        return VacationGrantRules(rules)
    }
}
